samples/fabric/fabric-cli/src/helpers/helpers.ts

```diff
@@ -572,29 +632,157 @@ const generateViewAddress = async (
+    if (addressParts.length != 6) {
+      throw new Error(`Expected 6 arguments for ${ccFunc}; found ${addressParts.length}`)
     }
-    if (addressParts[4] != sourceNetwork) {
-      throw new Error(`Passed source network ID ${sourceNetwork} does not match last chaincode argument in view address ${addressParts[4]}`)
+    if (addressParts[5] != sourceNetwork) {
+      throw new Error(`Passed source network ID ${sourceNetwork} does not match last chaincode argument in view address ${addressParts[5]}`)
     }
     const pledgeDetails = await getAssetPledgeDetails({
-      sourceNetworkName: addressParts[4],
+      sourceNetworkName: addressParts[5],
       pledger: '',
-      pledgerCert: addressParts[3],
+      pledgerCert: addressParts[4],
       destNetworkName: destNetwork,
       recipient: '',
-      recipientCert: addressParts[2],
-      assetType: addressParts[0],
-      assetRef: addressParts[1],
+      recipientCert: addressParts[3],
+      pledgeId: addressParts[0],
       logger: logger
     });
-    return viewAddress + ':' + JSON.parse(pledgeDetails).expirytimesecs;
+    return viewAddress + ':' + deserializeAssetPledge(pledgeDetails).getExpirytimesecs();
   } else {
     return viewAddress
   }
 }
 
+function deserializeAssetPledge(pledgeDetails) {
+    const pledgeDetailBytes = Buffer.from(pledgeDetails, 'base64')
+    const pledgeDetailsProto = AssetPledge.deserializeBinary(pledgeDetailBytes)
+    return pledgeDetailsProto
+}
+
+// Used for creating view address for interop call using remote-network-config.json
+const generateViewAddressFromRemoteConfig = (
+  networkId: string,
+  funcName: string,
+  funcArgs: Array<string>
+): any => {
+  const configPath = process.env.REMOTE_CONFIG_PATH
+    ? path.join(process.env.REMOTE_CONFIG_PATH)
+    : path.join(__dirname, '../../remote-network-config.json')
+  try {
+    const configJSON = JSON.parse(fs.readFileSync(configPath).toString())
+    if (!configJSON[networkId]) {
+      logger.error(
+        `Error: ${networkId} does not exist in the remote-network-config.json file`
+      )
+      throw new Error(`Error: ${networkId} does not exist in the remote-network-config.json file`)
+    }
+    
+    const remoteNetConfig = configJSON[networkId]
+    let address = remoteNetConfig.relayEndpoint + '/' + networkId
+    if (remoteNetConfig.type == "fabric") {
+        address = address + '/' + remoteNetConfig.channelName + ':' +
+            remoteNetConfig.chaincode + ':' + funcName + ':' + funcArgs.join(':');
+    } else if (remoteNetConfig.type == "corda") {
+        address = address + '/' + remoteNetConfig.PartyAEndPoint + '#' +
+            remoteNetConfig.flowPackage + "." + funcName + ":" + funcArgs.join(':');
+    } else {
+        logger.error(`Error: remote network ${remoteNetConfig.type} not supported.`)
+        throw new Error(`Error: remote network ${remoteNetConfig.type} not supported.`)
+    }
+    
+    return address
```

Comment:
Some lines end in semicolons and others don't. Can we keep this consistent at least within files?